src/js/navButtons.js

```diff
@@ -95,33 +85,32 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
     });
 
     gpii.firstDiscovery.navButtons.indexToDisposition = function (currentPanelNum, panelStartNum, panelTotalNum) {
-        return currentPanelNum === panelStartNum ? 0 : (currentPanelNum === panelTotalNum ? 2 : 1);
+        return currentPanelNum === panelStartNum ? 0 : (currentPanelNum < panelTotalNum - 1 ? 1 : 2);
     };
 
     gpii.firstDiscovery.navButtons.setButtonStates = function (that) {
         var currentPanelNum = that.model.currentPanelNum,
-            strings = that.options.strings,
             backButton = that.locate("back"),
             nextButton = that.locate("next"),
-            backButtonId = fluid.allocateSimpleId(backButton),
             nextButtonId = fluid.allocateSimpleId(nextButton),
             showSelector = that.options.styles.show,
             isFirstPanel = currentPanelNum === that.options.panelStartNum,
+            isLastPanel = currentPanelNum === that.options.panelTotalNum,
             disposition = gpii.firstDiscovery.navButtons.indexToDisposition(currentPanelNum, that.options.panelStartNum, that.options.panelTotalNum),
-            nextLabel = strings[["start", "next", "finish"][disposition]],
-            nextTooltipContent = strings[["startTooltip", "nextTooltip", "finishTooltip"][disposition]];
+            nextLabel = that.msgResolver.resolve(["start", "next", "finish"][disposition]),
```

Comment:
As discussed,
1. rename "gpii.firstDiscovery.messageResolver" to "gpii.firstDiscovery.msgLookup";
2. make use of the lookup functions from fluid.prefs.msgLookup rather than resolve() from the subcomponent, such as, replace that.msgResolver.resolve() with that.lookup()

Thanks